griptape/tokenizers/base_tokenizer.py

```diff
@@ -4,10 +4,22 @@
 from griptape import utils
 
 
-@define(frozen=True)
+@define()
 class BaseTokenizer(ABC):
+    MODEL_PREFIXES_TO_MAX_OUTPUT_TOKENS = {}
+    MODEL_PREFIXES_TO_MAX_TOKENS = {}
+
+    model: str = field(kw_only=True)
     stop_sequences: list[str] = field(default=Factory(lambda: [utils.constants.RESPONSE_STOP_SEQUENCE]), kw_only=True)
-    max_tokens: int = field(kw_only=True)
+    max_tokens: int = field(kw_only=True, default=None)
```

Comment:
`max_tokens` means _input_ tokens in Tokenizers, but `max_tokens` means _output_ tokens in Drivers. Should we rename this to `max_input_tokens`?

Reply:
I think we should!

Reply:
Agreed, I think it would be confusing otherwise. If we're making this naming change, though, should we rip the bandaid off and change `MODEL_PREFIXES_TO_MAX_TOKENS` to `MODEL_PREFIXES_TO_MAX_INPUT_TOKENS`?

Reply:
Done!